util/predicates/configmap_predicates.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ConfigMapCreate returns a predicate that returns true for a ConfigMap create event
+func ConfigMapCreateUpdate(logger logr.Logger) predicate.Funcs {
+	log := logger.WithValues("predicate", "ConfigMapCreateOrUpdate")
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			log = log.WithValues("eventType", "create")
+
+			c, ok := e.Object.(*corev1.ConfigMap)
+			if !ok {
+
+				log.V(4).Info("Expected ConfigMap", "type", e.Object.GetObjectKind().GroupVersionKind().String())
+				return false
+			}
+			log = log.WithValues("namespace", c.Namespace, "configmap", c.Name)
+
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log = log.WithValues("eventType", "update")
+
+			_, okOld := e.ObjectOld.(*corev1.ConfigMap)
+			newConfigMap, okNew := e.ObjectNew.(*corev1.ConfigMap)
+			if !okOld || !okNew {
+				log.V(4).Info("Expected ConfigMap", "type", e.ObjectOld.GetObjectKind().GroupVersionKind().String())
+				return false
+			}
+			log = log.WithValues("namespace", newConfigMap.Namespace, "configmap", newConfigMap.Name)
+
+			return true
```

Comment:
Seems like we're always returning true here? Should only return true if the resourceVersion, or generation changed instead?

Reply:
I think always true is fine - it's easier to read